lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -567,176 +547,226 @@ def write(chunk)
         log_name = "projects/#{@project_id}/logs/#{log_name(
           tag, group_level_resource)}"
 
-        # Does the actual write to the cloud logging api.
-        client = api_client
-        if @use_grpc
-          begin
-            labels_utf8_pairs = group_level_common_labels.map do |k, v|
-              [k.encode('utf-8'), convert_to_utf8(v)]
-            end
+        requests_to_send << {
+          entries: entries,
+          log_name: log_name,
+          resource: group_level_resource,
+          labels: group_level_common_labels
+        }
+      end
 
-            entries_count = entries.length
-            client.write_log_entries(
-              entries,
-              log_name: log_name,
-              resource: Google::Api::MonitoredResource.new(
-                type: group_level_resource.type,
-                labels: group_level_resource.labels.to_h
-              ),
-              labels: labels_utf8_pairs.to_h
-            )
-            increment_successful_requests_count
-            increment_ingested_entries_count(entries_count)
-
-            # Let the user explicitly know when the first call succeeded, to aid
-            # with verification and troubleshooting.
-            unless @successful_call
-              @successful_call = true
-              @log.info 'Successfully sent gRPC to Stackdriver Logging API.'
-            end
+      requests_to_send.each do |request|
+        @write_request.call(request)
```

Comment:
Should this be `**request`, to translate the hash into keyword arguments? Or does it "just work"(tm)?

Reply:
It just works because `request` is the only param here. `**request` is needed only if it's passing "additional" params, e.g.:

```
@write_request.call(entries: something_here, **request)
```